salt/modules/glusterfs.py

```diff
@@ -43,34 +43,46 @@ def _get_minor_version():
     return version
 
 
-def _gluster(cmd):
+def _gluster_ok(xml_data):
     '''
-    Perform a gluster command.
+    Extract boolean return value from Gluster's XML output.
     '''
-    # We will pass the command string as stdin to allow for much longer
-    # command strings. This is especially useful for creating large volumes
-    # where the list of bricks exceeds 128 characters.
-    return __salt__['cmd.run'](
-        'gluster --mode=script', stdin="{0}\n".format(cmd))
+    return int(xml_data.find('opRet').text) == 0
 
```

Comment:
There are a few gluster commands that don't export xml. The _gluster command needs to remain to handle those exceptions.